[ofono] call SetDefaultDataSim on Default Data Sim change

natur-produkt opened the pull request:
If Default Data Sim is changed,
then we need to call SetDefaultDataSim from
ofono service to enable updated RIL

mardy commented on src/indicator/nmofono/manager-impl.cpp:
```diff
@@ -479,6 +479,24 @@ public Q_SLOTS:
+
+        if (sim != nullptr){
+            QDBusConnection bus = QDBusConnection::systemBus();
+            if (!bus.isConnected())
```
`assert(bus.isConnected());`

natur-produkt replied:
true - changed in next push

mardy commented on src/indicator/nmofono/manager-impl.cpp:
```diff
@@ -479,6 +479,24 @@ public Q_SLOTS:
+                                      "org.nemomobile.ofono.ModemManager",
+                                      bus);
+
+            if ( dbus_iface.isValid() ) {
```
We don't use spaces around the `if` expressions. But in this case I think that this check is not needed: even if the interface is not valid, QtDBus will just return an error (which we are ignoring).

mardy commented on src/indicator/nmofono/manager-impl.cpp:
```diff
@@ -479,6 +479,24 @@ public Q_SLOTS:
+            if ( dbus_iface.isValid() ) {
+                QDBusPendingReply<QVariantMap> reply = dbus_iface.asyncCall("SetDefaultDataSim",
+                                                                m_simForMobileData->imsi());
+                reply.waitForFinished();
```
There's no point in making an async call if you are waiting for its result anyway.

natur-produkt replied:
true, I changed this

mardy reviewed (changes requested):
Hi, and thanks for this!
  But I don't think that this is the right place for this change: this class does not deal directly with the Ofono D-Bus interface, but rather delegates the work to the classes under the `wwan/` subdirectory. I don't know this project well, but my guess is that you should add a method to set the default sim to the `wwan::SimManager` class and call it from here.

natur-produkt commented:
Thank you very much for your review - I really appreciate it. I'm new in project, so any feedback is needed. I changed the logic a little bit, added signal to manager-impl.cpp, added slot to sim-manager.cpp and connected those two.

I know that it can be not a final version, but waiting for your review!

mariogrip reviewed (approved):
LGTM :+1:

mariogrip commented:
> Thank you very much for your review - I really appreciate it. I'm new in project, so any feedback is needed. I changed the logic a little bit, added signal to manager-impl.cpp, added slot to sim-manager.cpp and connected those two.
> 
> I know that it can be not a final version, but waiting for your review!

This is fine for now, but it will need to be cleaned a bit up before merging into the "xenial" branch, but for now this works great.
